koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/BuildList.java

```diff
@@ -0,0 +1,80 @@
+package org.fakekoji.xmlrpc.server.xmlrpcresponse;
+
+import hudson.plugins.scm.koji.Constants;
+import hudson.plugins.scm.koji.model.Build;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.fakekoji.xmlrpc.server.xmlrpcrequestparams.XmlRpcRequestUtils.toMaps;
+
+public class BuildList implements XmlRpcResponse<List<Build>> {
+
+    private final List<Build> builds;
+
+    public BuildList(List<Build> builds) {
+        this.builds = builds;
+    }
+
+    public BuildList(Object object) {
+        builds = parseBuildMaps(toMaps(object));
+    }
+
+    @Override
+    public Object toObject() {
+        return parseBuilds();
+    }
+
+    @Override
+    public List<Build> getValue() {
+        return builds;
+    }
+
+    private List<Build> parseBuildMaps(List<Map<String, Object>> maps) {
+        if (maps == null) {
+            return Collections.emptyList();
+        }
+        final List<Build> builds = new ArrayList<>(maps.size());
+        for (Map<String, Object> map : maps) {
+            builds.add(parseBuildMap(map));
+        }
+        return builds;
+    }
+
+    private Build parseBuildMap(Map<String, Object> map) {
+        return new Build(
+                (Integer) map.get(Constants.build_id),
+                (String) map.get(Constants.name),
+                (String) map.get(Constants.version),
+                (String) map.get(Constants.release),
+                (String) map.get(Constants.nvr),
+                (String) map.get(Constants.completion_time),
+                null,
+                null,
+                null
+        );
+    }
+
+    private List<Map<String, Object>> parseBuilds() {
+        List<Map<String, Object>> maps = new ArrayList<>();
+        for (Build build : builds) {
+            maps.add(parseBuild(build));
+        }
+        return maps;
+    }
+
+    private Map<String, Object> parseBuild(Build build) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(Constants.name, build.getName());
+        map.put(Constants.version, build.getVersion());
+        map.put(Constants.release, build.getRelease());
+        map.put(Constants.nvr, build.getNvr());
+        map.put(Constants.build_id, build.getId());
+        map.put(Constants.completion_time, build.getCompletionTime());
+        // map.put(Constants.rpms, parseRpms(build.getRpms()));
```

Comment:
why is this commented out? can we remove it?

Reply:
I forgot to remove it. RPMs are retrieved in getRPMs method, not in listBuilds